substrate/runtime/system/src/lib.rs

```diff
@@ -68,6 +75,7 @@ decl_storage! {
 	pub BlockHash get(block_hash): b"sys:old" => required map [ T::BlockNumber => T::Hash ];
 
 	pub ExtrinsicIndex get(extrinsic_index): b"sys:xti" => required u32;
+	pub ExtrinsicData get(extrinsic_data): b"sys:xtd" => required map [ u32 => Vec<u8> ];
```

Comment:
shouldn't this be cleared in `finalise`? all the values are `take`n out of it in `derive_extrinsics` but it probably leaves some residuals like length in storage.

Reply:
shouldn't be possible.

Reply:
it's both notionally and logically impossible. furthermore there's no way of definitively clearing a mapping.

Reply:
a later PR that (re-)instates an "list" type might be reasonable though.

Reply:
I don't think it's logically impossible as soon as mappings get any kind of metadata keys (for example, to enable iteration). This is something we will have to watch out for.